sdk/cosmos/src/clients/database_client.rs

```diff
@@ -91,8 +94,85 @@ impl DatabaseClient {
     }
 
     /// List users
-    pub fn list_users(&self) -> requests::ListUsersBuilder<'_, '_> {
-        requests::ListUsersBuilder::new(self)
+    pub fn list_users(
+        &self,
+        ctx: Context,
+        options: ListUsersOptions,
+    ) -> impl Stream<Item = Result<ListUsersResponse, crate::Error>> + '_ {
+        macro_rules! r#try {
```

Comment:
We'll want to move most of this logic to a shared place once this and #340 lands.